File: DONGUK/프로그래머스_WEEK 5/배달.py
import math

INF = math.inf

def solution(N, road, K):
    answer = 0
    graph = [[INF] * (N+1) for _ in range(N+1)]
    
    for data in road:
        start, end, dis = data[0], data[1], data[2]
        
        if graph[start][end] != INF:
            graph[start][end] = min(graph[start][end], dis)
            
        else:
            graph[start][end] = dis
        
        if graph[end][start] != INF:
            graph[end][start] = min(graph[end][start], dis)
            
        else:
            graph[end][start] = dis
    
    
    for k in range(1, N+1):
        for i in range(1, N+1):
            for j in range(1, N+1):
                if i == j:
                    graph[i][j] = 0
                    continue
                graph[i][j] = min(graph[i][j], graph[i][k] + graph[k][j])
                
    
    for num in graph[1]:
        if num <= K:
            answer += 1

    return answer

File: KIJUNG/PRO_WEEK_06/배달.cpp
#include <iostream>
#include <vector>
#include <algorithm>

#define INF 1000000000

using namespace std;

int solution(int N, vector<vector<int> > road, int K) {
    int answer = 1;
    int city[51][51];
    fill(&city[0][0], &city[50][51], INF);
    for(vector<int> v: road){
        city[v[0]][v[1]] = min(v[2],city[v[0]][v[1]]);
        city[v[1]][v[0]] = min(v[2],city[v[1]][v[0]]);
    } 

    for(int k=1;k<N+1;k++){
        for(int i=1;i<N+1;i++){
            for(int j=1;j<N+1;j++){
                if(city[i][k] != INF && city[k][j] != INF){
                    city[i][j] = min(city[i][j], city[i][k] + city[k][j]);
                }
            }
        }
    }

    for(int i=2;i<N+1;i++){
        if(city[1][i] <= K) answer++;
    }

    return answer;
}